lib/puppet/provider/reboot_notify/notify.rb

```diff
@@ -45,9 +53,9 @@ def self.post_resource_eval
     target = "#{Puppet[:vardir]}/reboot_notifications.json"
     records = {}
     begin
-      records = PSON.parse(File.read(target))
+      records = JSON.parse(File.read(target))
     rescue
```

Comment:
Would rather see rescue of exceptions raised by File.read() and JSON.parse() and reporting of the error these exceptions provide. . For example,

raise JSON::JSONError,ArgumentError,TypeError,SystemCallError => e
   raise(Puppet::Error, "Could not parse file: #{target} - #{e.message}")